version: '3'

services:

  reverse-proxy:
    image: traefik:v2.6
    ports:
      - "80:80"
      - "8088:8080"
      - "443:443"
    environment:
      - AWS_ACCESS_KEY_ID
      - AWS_SECRET_ACCESS_KEY
      - AWS_HOSTED_ZONE_ID
      - AWS_REGION
      - EMAIL
    volumes:
      - "/var/letsencrypt:/letsencrypt"
      - "/var/traefik:/etc/traefik"
    networks:
      - traefik_public
      - socket_proxy
    restart: unless-stopped
    depends_on:
      - socket_proxy
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik.rule=Host(`${TRAEFIK_HOST}`)"
      - "traefik.http.routers.traefik.entrypoints=websecure"
      - "traefik.http.routers.traefik.service=api@internal"
      - "traefik.http.routers.traefik.middlewares=auth"
      - "traefik.http.middlewares.auth.digestauth.usersfile=/etc/traefik/users"
      
  socket_proxy:
    image: tecnativa/docker-socket-proxy:latest
    restart: unless-stopped
    environment:
      CONTAINERS: 1
      NETWORKS: 1
      SERVICES: 1
      TASKS: 1
      LOG_LEVEL: warning
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    networks:
      - socket_proxy
      
networks:
  traefik_public:
    external: true
  socket_proxy:
    external: true
